controllers/factory_test.go

```diff
@@ -52,6 +52,26 @@ func TestFactoryBuildAllDockerReconciler(t *testing.T) {
 	g.Expect(reconcilers.DockerDatacenterReconciler).NotTo(BeNil())
 }
 
+func TestFactoryBuildAllTinkerbellReconciler(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	logger := nullLog()
+	ctrl := gomock.NewController(t)
+	manager := mocks.NewMockManager(ctrl)
+	manager.EXPECT().GetClient().AnyTimes()
+	manager.EXPECT().GetScheme().AnyTimes()
+
+	f := controllers.NewFactory(logger, manager).
+		WithTinkerbellDatacenterReconciler()
+
+	// testing idempotence
+	f.WithTinkerbellDatacenterReconciler()
+
+	reconcilers, err := f.Build(ctx)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(reconcilers.TinkerbellDatacenterReconciler).NotTo(BeNil())
```

Comment:
How is this testing idemptotence? Would the factory `Build` method return an err if there is more than one `WithTinkerbellDatacenterReconciler` registered?

Reply:
yes it would.